File: c/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;
    float index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

int count_letters(string text)
{
    int letters = 0;
    for (int letter = 0, n = strlen(text); letter < n; letter++)
    {
        if (isalpha(text[letter]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 0;
    for (int letter = 0, n = strlen(text); letter < n; letter++)
    {
        if (isspace(text[letter]))
        {
            words++;
        }
    }
    return words + 1;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int letter = 0, n = strlen(text); letter < n; letter++)
    {
        if (text[letter] == '.' || text[letter] == '!' || text[letter] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: python/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = letters / words * 100
    S = sentences / words * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index < 1:
        print("Before Grade 1")

    elif index >= 16:
        print("Grade 16+")

    else:
        print("Grade", round(index))


def count_letters(text):
    letters = 0

    for letter in text:
        if letter.isalpha():
            letters += 1

    return letters


def count_words(text):
    words = 0

    for letter in text:
        if letter.isspace():
            words += 1

    return words + 1


def count_sentences(text):
    sentences = 0

    for letter in text:
        if letter == "." or letter == "!" or letter == "?":
            sentences += 1

    return sentences


if __name__ == "__main__":
    main()
